### PR TITLE
CAUTH-32 Add support for GET

### PR DESCRIPTION
**Pull request by kusold**
This adds client support for the `GET` definition in limitd/protocol#3

### CODE REVIEW

File: client.js [hunk 1]
```diff
@@ -450,6 +450,29 @@ LimitdClient.prototype.put = function (type, key, count, done) {
   return this._retriedRequest(request, done);
 };
 
+LimitdClient.prototype.get = function (type, key, done) {
+  const fireAndForget = typeof done !== 'function';
+
+  if (typeof key !== 'string') {
+    // TODO: Decide if I should error
```
[the remark this answers is not in the document]
Comment (kusold) [reply client.js]: We are letting the server take care of that detail: https://github.com/limitd/limitdb/pull/11/files#diff-be228df81b99a384aa50ad50bde7fca0R441

File: client.js [hunk 2]
```diff
@@ -450,6 +450,29 @@ LimitdClient.prototype.put = function (type, key, count, done) {
...
+  if (fireAndForget) {
+    return this._fireAndForgetRequest(request);
+  }
+
+  return this._retriedRequest(request, done);
+};
+
 LimitdClient.prototype.status = function (type, key, done) {
   if (typeof key !== 'string') {
     key = '';
```
[the remark this answers is not in the document]
Comment (kusold) [reply client.js]: Rejected on the server if there is no key. 👍

File: client.js [hunk 3]
```diff
@@ -450,6 +450,28 @@ LimitdClient.prototype.put = function (type, key, count, done) {
   return this._retriedRequest(request, done);
 };
 
+LimitdClient.prototype.get = function (type, key, done) {
+  const fireAndForget = typeof done !== 'function';
+
+  if (typeof key !== 'string') {
```
Comment (dafortune) [inline client.js]: Why?

Comment (dafortune) [reply client.js]: I mean why this is not an error

File: test/shard_client.tests.js [hunk 4]
```diff
@@ -122,6 +122,28 @@ describe('ShardClient', function() {
...
+    };
+
+    const ShardClient = ShardClientCtor(client);
+
+    const shardClient = new ShardClient({
+      client,
+      shard: { hosts: [ 'host-1', 'host-2' ] }
+    });
+
+    shardClient.get('ip', '10.0.0.1', _.noop);
```
Comment (dafortune) [inline test/shard_client.tests.js]: Check that the response is actually what you expect?

File: client.js [hunk 5]
```diff
@@ -450,6 +450,28 @@ LimitdClient.prototype.put = function (type, key, count, done) {
   return this._retriedRequest(request, done);
 };
 
+LimitdClient.prototype.get = function (type, key, done) {
+  const fireAndForget = typeof done !== 'function';
```
Comment (dafortune) [inline client.js]: Why would you call a getter as fire and forget?

Comment (kusold) [reply client.js]: Removed the fire and forget option. I was following the pattern that the other methods used, but you're right that it doesn't make sense to call `get` that way.

File: test/shard_client.tests.js (hunk 4 above)
Comment (kusold) [reply test/shard_client.tests.js]: Added. 👍

File: client.js (hunk 3 above)
Comment (kusold) [reply client.js]: Changed to error. 👍

File: package.json [hunk 6]
```diff
@@ -15,7 +15,7 @@
     "hashring": "^3.2.0",
     "length-prefixed-message": "^3.0.3",
     "length-prefixed-stream": "~1.4.0",
-    "limitd-protocol": "limitd/protocol#v2.0.1",
+    "limitd-protocol": "kusold/protocol#get",
```
Comment (kusold) [inline package.json]: Waiting for a new release to get cut after limitd/protocol#3 is merged

**Pull request merged**